l10n_be_national_number/migrations/16.0.2.0.0/pre-migration.py

```diff
@@ -0,0 +1,17 @@
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    old_id = env.ref("l10n_be_national_number.l10n_be_national_number_category")
+    new_id = env.ref(
+        "l10n_be_partner_identification.l10n_be_national_registry_number_category"
+    ).id
+    for id_numbers in (
+        env["res.partner"]
+        .with_context(active_test=False)
+        .search([])
+        .mapped("id_numbers")
+    ):
+        if id_numbers.category_id == old_id:
+            id_numbers.category_id = new_id
```

Comment:
actually, i think this won’t work if there are multiple `id_numbers` on a partner, which can be the case as its a one2many relationship. instead, you should work with `res.partner.id_number` records directly. one way is to do it like this:

```suggestion
    old_id = env.ref("l10n_be_national_number.l10n_be_national_number_category").id
    new_id = env.ref(
        "l10n_be_partner_identification.l10n_be_national_registry_number_category"
    ).id
    id_numbers = (
        env["res.partner.id_number"]
        .with_context(active_test=False)
        .search([("category_id", "=", old_id)])
    )
    id_numbers.write({"category_id": new_id})
```

another is like this:

```suggestion
    old_id = env.ref("l10n_be_national_number.l10n_be_national_number_category").id
    new_id = env.ref(
        "l10n_be_partner_identification.l10n_be_national_registry_number_category"
    ).id
    env.cr.execute(
        """
        update res_partner_id_number
        set category_id = %(old_id)s
        where category_id = %(new_id}s
        """,
        {"old_id": old_id, "new_id": new_id},
    )
```